lib/+wbs/@Robot/Robot.m

```diff
@@ -63,91 +52,61 @@ function set_robot_state(obj, w_H_b, s, base_pose_dot, s_dot)
             %        - s = [NDOF, 1] Joints position vector
             %        - base_pose_dot = [6,1] linear and angular velocity of the base
             %        - s_dot = [NDOF, 1] Joints velocity vector
-            iDynTreeWrappers.setRobotState(obj.KinDynModel, w_H_b, s, ...
-                base_pose_dot, s_dot, obj.g);
+            obj.KinDynModel.kinDynComp.setRobotState(w_H_b, s, base_pose_dot, s_dot);
         end
 
         function M = get_mass_matrix(obj,motorInertias)
             % get_mass_matrix Returns the mass matrix
             % OUTPUT: - M: mass matrix
-            if (~obj.KinDynModel.kinDynComp.getFreeFloatingMassMatrix(obj.M_iDyn))
+            [ack,M] = obj.KinDynModel.kinDynComp.getFreeFloatingMassMatrix();
+            if (~ack)
                 error('[Robot: get_mass_matrix] Unable to retrieve the mass matrix');
             end
 
             % Add the reflected inertia if the feature is activated
             if obj.useMotorReflectedInertias
-                M = obj.M_iDyn.toMatlab + diag([zeros(6,1);motorInertias]);
-            else
-                M = obj.M_iDyn.toMatlab;
+                M = M + diag([zeros(6,1);motorInertias]);
             end
         end
 
         function h = get_bias_forces(obj)
             % get_bias_forces Returns the bias force
             % OUTPUT: - h: bias force
-            if (~obj.KinDynModel.kinDynComp.generalizedBiasForces(obj.h_iDyn))
+            [ack,h] = obj.KinDynModel.kinDynComp.generalizedBiasForces();
+            if (~ack)
                 error('[Robot: get_bias_forces] Unable to retrieve the bias force')
             end
-
-            h_b = obj.h_iDyn.baseWrench.toMatlab;
-            h_s = obj.h_iDyn.jointTorques.toMatlab;
-            h = [h_b; h_s];
         end
 
         function [J_LFoot, J_RFoot] = get_feet_jacobians(obj)
             % get_feet_jacobians Returns the Jacobians of the feet
             % OUTPUT: - J_left_foot: Jacobian of the left foot
             %         - J_right_foot: Jacobian of the right foot
-            if (~obj.KinDynModel.kinDynComp.getFrameFreeFloatingJacobian(obj.LFoot_frameID, obj.J_LFoot_iDyntree))
+            [ack,J_LFoot] = obj.KinDynModel.kinDynComp.getFrameFreeFloatingJacobianLRfoot('LFoot');
+            if (~ack)
                 error('[Robot: get_feet_jacobians] Unable to retrieve the left foot jacobian');
             end
 
-            if (~obj.KinDynModel.kinDynComp.getFrameFreeFloatingJacobian(obj.RFoot_frameID, obj.J_RFoot_iDyntree))
+            [ack,J_RFoot] = obj.KinDynModel.kinDynComp.getFrameFreeFloatingJacobianLRfoot('RFoot');
+            if (~ack)
                 error('[Robot: get_feet_jacobians] Unable to retrieve the left foot jacobian');
```

Comment:
Typo: left -> right

Reply:
bad copy paste :D
Fixed in next commit.